.travis.yml

```diff
@@ -0,0 +1,53 @@
+# This is the simple Travis configuration, which is intended for use
+# on applications which do not require cross-platform and
+# multiple-GHC-version support. For more information and other
+# options, see:
+#
+# https://docs.haskellstack.org/en/stable/travis_ci/
+#
+# Copy these contents into the root directory of your Github project in a file
+# named .travis.yml
+
+# Use new container infrastructure to enable caching
+sudo: false
+
+# Do not choose a language; we provide our own build tools.
+language: generic
+
+# Caching so the next build will be fast too.
+cache:
+  directories:
+  - $HOME/.stack
+
+# Ensure necessary system libraries are present
+addons:
+  apt:
+    packages:
+      - libgmp-dev
+
+before_install:
+# Download and unpack the stack executable
+- mkdir -p ~/.local/bin
+- export PATH=$HOME/.local/bin:$PATH
+- travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+- travis_retry curl -L -O https://github.com/TorXakis/Dependencies/releases/download/v0.3.0_linux/z3-4.5.1.0f1583309d08-x64-ubuntu-14.04.zip -O https://github.com/TorXakis/Dependencies/releases/download/v0.3.0_linux/cvc4-1.5-x86_64-linux-opt
+# rename cvc4 executable to get the build working todo: parametrize this in SMTInternal.hs
+- mv cvc4-1.5-x86_64-linux-opt cvc4-2017-06-13-win32-opt

```

Comment:
Something to do in the future: make a bash/haskell script to install cvc4 and z3.